libs/server/src/lib/utils/utils.ts

```diff
@@ -121,21 +125,36 @@ export function listFiles(dirName: string): string[] {
 export function directoryExists(filePath: string): boolean {
   try {
     return statSync(filePath).isDirectory();
-  } catch (err) {
+  } catch {
     return false;
   }
 }
 
 export function fileExistsSync(filePath: string): boolean {
   try {
     return statSync(filePath).isFile();
-  } catch (err) {
+  } catch {
     return false;
   }
 }
 
-function readAndParseJson(fullFilePath: string): any {
-  return JSON.parse(stripJsonComments(readFileSync(fullFilePath).toString()));
+function readAndParseJson(filePath: string) {
+  const content = readFileSync(filePath, 'utf-8');
+  try {
+    return JSON.parse(content);
+  } catch {
+    const errors: ParseError[] = [];
+    const result = parseJson(content, errors);
+
+    if (errors.length > 0) {
+      const { error, offset } = errors[0];
+      throw new Error(
+        `${printParseErrorCode(error)} in JSON at position ${offset}`
+      );
+    }
+
+    return result;
```

Comment:
This is neat, I have to run a few performance tests locally to see how well this performs. The repo I do for this is here if you want to try it yourself as well:
https://github.com/Cammisuli/nx-incremental-large-repo/tree/1425-projects

There's essentially 1425 projects, and we need to make sure that we can parse this fairly quickly to get a list of all the projects, and their targets.